File: library/src/main/java/zjm/cst/dhu/library/inject/module/ApplicationModule.java
package zjm.cst.dhu.library.inject.module;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import zjm.cst.dhu.library.MyApplication;
import zjm.cst.dhu.library.inject.scope.PerApplication;
import zjm.cst.dhu.library.inject.scope.PerContext;

/**
 * Created by zjm on 2017/5/9.
 */
@Module
public class ApplicationModule {
    private MyApplication myApplication;

    public ApplicationModule(MyApplication myApplication) {
        this.myApplication = myApplication;
    }

    @Provides
    @PerApplication
    @PerContext("Application")
    public Context provideMyApplicationContext() {
        return myApplication.getApplicationContext();
    }
}

File: library/src/main/kotlin/zjm/cst/dhu/library/inject/module/ApplicationModule.kt
package zjm.cst.dhu.library.inject.module

import android.content.Context
import dagger.Module
import dagger.Provides
import zjm.cst.dhu.library.MyApplication
import zjm.cst.dhu.library.inject.scope.PerApplication
import zjm.cst.dhu.library.inject.scope.PerContext

/**
 * Created by zjm on 2017/5/9.
 */
@Module
class ApplicationModule(private val myApplication: MyApplication) {

    @Provides
    @PerApplication
    @PerContext("Application")
    fun provideMyApplicationContext():
            Context = myApplication.applicationContext
}
